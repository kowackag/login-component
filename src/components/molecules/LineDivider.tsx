import { HorizontalLine } from "../atoms/HorizontalLine";

export const LineDivider = () => {
	return (
		<div className="my-3 flex items-center">
			<HorizontalLine />
			<p className="font-primaryRegular rounded-md  border-2 border-neutral-300 px-1 py-0.5 uppercase text-neutral-400">
				or
			</p>
			<HorizontalLine />
		</div>
	);
};
