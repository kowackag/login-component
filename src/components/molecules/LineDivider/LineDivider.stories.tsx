import type { Meta, StoryObj } from "@storybook/react";

import { LineDivider } from "./LineDivider";

const meta: Meta<typeof LineDivider> = {
	component: LineDivider,
};

export default meta;

type Story = StoryObj<typeof LineDivider>;

export const Default: Story = {
	args: {},
};
