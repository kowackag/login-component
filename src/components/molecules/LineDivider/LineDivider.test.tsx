import { expect, describe, it } from "vitest";
import { render, screen } from "@testing-library/react";
import { LineDivider } from "./LineDivider";

describe("LineDivider ", () => {
	it("has two horizontal line", () => {
		render(<LineDivider />);
		const hrLines = screen.getAllByTestId("hrLine");
		expect(hrLines.length).toBe(2);
	});
	it("renders correctly", () => {
		const wrapper = render(<LineDivider />);
		expect(wrapper.container).toMatchSnapshot();
	});
});
