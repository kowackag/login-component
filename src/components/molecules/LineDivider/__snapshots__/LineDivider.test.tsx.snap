// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LineDivider  > renders correctly 1`] = `
<div>
  <div
    class="my-3 flex items-center"
  >
    <span
      class="flex-grow border border-b border-b-gray-300"
      data-testid="hrLine"
    />
    <p
      class="rounded-md border-2  border-neutral-300 px-1 py-0.5 font-primaryRegular uppercase text-neutral-400"
    >
      or
    </p>
    <span
      class="flex-grow border border-b border-b-gray-300"
      data-testid="hrLine"
    />
  </div>
</div>
`;
